test/optimizer/operator_transformer_test.cpp

```diff
@@ -841,7 +841,7 @@ TEST_F(OperatorTransformerTest, SubqueryComplexTest) {
 
   std::string select_sql =
       "SELECT A.a1 FROM A WHERE A.a1 IN (SELECT b1 FROM B WHERE b1 = 2 AND "
-      "b2 > (SELECT a1 FROM A WHERE a2 > 0)) AND EXISTS (SELECT b1 FROM B WHERE B.b1 = A.a1)";
+      "b1 > (SELECT a1 FROM A WHERE a1 > 0)) AND EXISTS (SELECT b1 FROM B WHERE B.b1 = A.a1)";
```

Comment:
In particular, 
```
ERROR:  operator does not exist: character varying > integer
LINE 1: ... where b1 = 2 and b2 > (select a1 from A where a2 > 0)) and ...
                                                                                                 ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
```
and
```
ERROR:  operator does not exist: character varying > integer
LINE 1: ... where b1 = 2 and b2 > (select a1 from A where a2 > 0)) and ...
                                                  ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
```

@mush-zhang Can you confirm that this test is fine? I am not sure what the original test was meant to test.

Reply:
it's testing if the subqueries are processed correctly. Previous binder does not have strict type checking or any type inferring thing. So it is not catching some sematically incorrect statements.
maybe change it to 
`"SELECT A.a1 FROM A WHERE A.a1 IN (SELECT b1 FROM B WHERE b1 = 2 AND "
      "b1 > (SELECT MAX(a1) FROM A WHERE a2 > 0)) AND EXISTS (SELECT b1 FROM B WHERE B.b1 = A.a1)";`

Reply:
I realize the formatting got a bit messed up, but the `a2 > 0` is the issue - `a2` is a varchar. I have already changed it to be something that typechecks. Does the current version test what you originally intended to test?